pyomo/solvers/plugins/solvers/xpress_direct.py

```diff
@@ -436,98 +436,96 @@ def _get_nlp_results(self, results, soln):
-            elif status == xp.nlp_unfinished:
-                results.solver.status = SolverStatus.ok
-                results.solver.termination_message = (
-                    "Non-convex solve not finished (numerical issues?)"
+                results.solver.termination_condition = (
+                    TerminationCondition.locallyOptimal
                 )
-                results.solver.termination_condition = TerminationCondition.unknown
-                soln.status = SolutionStatus.unknown
-                have_soln = True
+                soln.status = SolutionStatus.locallyOptimal
             else:
-                results.solver.status = SolverStatus.error
+                results.solver.status = SolverStatus.ok
                 results.solver.termination_message = (
-                    "Error for non-convex model: " + str(status)
+                    "Feasible solution found for non-convex model"
                 )
-                results.solver.termination_condition = TerminationCondition.error
-                soln.status = SolutionStatus.error
+                results.solver.termination_condition = TerminationCondition.feasible
+                soln.status = SolutionStatus.feasible
+            have_soln = True
+        elif status == xp.nlp_globally_optimal:
+            results.solver.status = SolverStatus.ok
+            results.solver.termination_message = (
+                "Non-convex model was solved to global optimality"
+            )
+            results.solver.termination_condition = TerminationCondition.optimal
+            soln.status = SolutionStatus.optimal
+            have_soln = True
+            optimal = True
+        elif status == xp.nlp_locally_infeasible:
+            results.solver.status = SolverStatus.ok
+            results.solver.termination_message = (
+                "Non-convex model was proven to be locally infeasible"
+            )
+            results.solver.termination_condition = TerminationCondition.noSolution
+            soln.status = SolutionStatus.unknown
+        elif status == xp.nlp_infeasible:
+            results.solver.status = SolverStatus.ok
+            results.solver.termination_message = (
+                "Non-conex model was proven to be infeasible"
+            )
+            results.solver.termination_condition = TerminationCondition.infeasible
+            soln.status = SolutionStatus.infeasible
+        elif status == xp.nlp_unbounded:  # locally unbounded!
+            results.solver.status = SolverStatus.ok
+            results.solver.termination_message = "Non-convex model is locally unbounded"
+            results.solver.termination_condition = TerminationCondition.unbounded
+            soln.status = SolutionStatus.unbounded
+        elif status == xp.nlp_unfinished:
+            results.solver.status = SolverStatus.ok
+            results.solver.termination_message = (
+                "Non-convex solve not finished (numerical issues?)"
```

Comment:
Please remind me later to change this error message. Because it's entirely not helpful.